packages/amplify-category-auth/src/provider-utils/supported-services.ts

```diff
@@ -1164,6 +1176,284 @@ export const supportedServices = {
+            operator: "exists"
+          },
+          {
+            onCreate: "never"
+          },
+        ]
+      },
+      {
+        key: "EditMappings",
+        question: "Which mappings do you want to edit ?",
+        type: "multiselect",
+        iterator: "oidcAttributesMapping",
+        iteratorValidation: {
+          operator: "regex",
+          value: ".*",
+          onErrorMsg: "The value must be a valid string."
+        },
+        andConditions: [
+          {
+            key: "oidcAttributesMapping",
+            operator: "exists"
+          },
+          {
+            onCreate: "never"
+          },
+          {
+            key: "EditMappingConfirmation",
+            value: true,
+            operator: "="
+          }
+        ]
+      },
+      {
+        key: "RemoveMappingConfirmation",
+        question: "Do you want to remove some attributes mapping ?",
+        type: "confirm",
+        andConditions: [
+          {
+            key: "authProvidersUserPool",
+            value: "OIDC",
+            operator: "includes"
+          },
+          {
+            key: "oidcAttributesMapping",
+            operator: "exists"
+          },
+          {
+            onCreate: "never"
+          },
+        ]
+      },
+      {
+        key: "RemoveMappings",
+        question: "Which mappings do you want to remove ?",
+        type: "multiselect",
+        iterator: "oidcAttributesMapping",
+        iteratorValidation: {
+          operator: "regex",
+          value: ".*",
+          onErrorMsg: "The value must be a valid string."
+        },
+        andConditions: [
+          {
+            key: "oidcAttributesMapping",
+            operator: "exists"
+          },
+          {
+            onCreate: "never"
+          },
+          {
+            key: "RemoveMappingConfirmation",
+            value: true,
+            operator: "="
+          }
+        ]
+      },
+      {
+        key: "OIDCAdditionalMapping",
+        question: "Do you want to add specific attribute mapping ?",
+        type: "confirm",
+        andConditions: [
+          {
+            key: "authProvidersUserPool",
+            value: "OIDC",
+            operator: "includes"
+          },
+        ]
+      },
+      {
+        key: "newOIDCMapping",
+        question: "Which mappings do you want to add ?",
+        required: true,
+        type: "multiselect",
+        map: "coreAttributes",
+        iterator: "oidcAttributesMapping",
+        requiredOptions: ["usernameAttributes", "requiredAttributes"],
+        andConditions: [
+          {
+            key: "OIDCAdditionalMapping",
+            value: true,
+            operator: "="
+          },
+          {
+            preventEdit: 'exists',
+            key: 'oidcAttributesMapping',
+          },
+          {
+            preventEdit: 'existsInCurrent',
+            key: 'oidcAttributesMapping',
+          },
+        ]
+      },
+  {
+
+          key: "newOIDCMapping",
```

Comment:
Why do we have two keys with the same name: newOIDCMapping?

Reply:
those are same questions but just part of different flow (add vs update). I can change the key to "newOIDCMappingOnUpdate" if it's clearer ?